packages/data/src/types.ts

```diff
@@ -170,9 +170,17 @@ export type ConfigOf< S > = S extends StoreDescriptor< infer C > ? C : never;
 
 export type ActionCreatorsOf< Config extends AnyConfig > =
 	Config extends ReduxStoreConfig< any, infer ActionCreators, any >
-		? ActionCreators
+		? PromisifiedActionCreator< ActionCreators >
 		: never;
 
+// When dispatching an action creator, the return value is a promise.
+type PromisifiedActionCreator< ActionCreators extends MapOf< ActionCreator > > =
```

Comment:
Might be worth exporting the new types in order to be able to consume them externally. Including the singular `PromisifiedActionCreator` and the plural `PromisifiedActionCreators` that Jarda suggested.

Reply:
Done!